### PR TITLE
[16.0][MIG] - rest_log

### PR DESCRIPTION
**Pull request by sbejaoui**

### CODE REVIEW

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
@sbejaoui Can you prepare a PR base on this one with the improvements listed below:
* https://github.com/acsone/rest-framework/commit/d542d31d7c60950a32c272d9612253f3c887481a
* https://github.com/acsone/rest-framework/commit/3f84ffa477116f4aae7d95628c6e8a536d03d4bb
* https://github.com/acsone/rest-framework/commit/1f750a4c63ade66a0ee78779d533db8532ba3cd4
* https://github.com/acsone/rest-framework/commit/35f06b977d5aa71edbb1a1f1db2f530f4308567c
* https://github.com/acsone/rest-framework/commit/1032dfdf3eb4442e5d39fb7ae8efe050b5fdbea1

These come from https://github.com/acsone/rest-framework/commits/10.0-backport-from-14.0/rest_log

**Review by lmignon — APPROVED** [review]
LGTM (Code review)

**Comment by sbidoul** [comment]
We need to finish #360 to get the 16.0 branch green again.

File: test-requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 httpx
+mock
```
Comment (simahawk) [inline test-requirements.txt]: mock is not required, pls import it from unittest

**Review by simahawk — CHANGES_REQUESTED** [review]
I see some fixes: can you please check v14?
AFAIR we made some fix/imp there (the Response obj for instance).

**Comment by lmignon** [comment]
> 

@sbejaoui Can you include:
* 7b3e019a31ef620f3433a75dc47b8af8df5c4e26
* 41ff3aeac023e899bd5778728f87826f9e69ec62
* 6db08371025d449337b9621ee6aed75b8518e877
* 5d6ee6d42b698bb681ac0409a4136261b7e7eba1
* 89ae7cc8febf1ad01c994fa7801f723f6cdd732c
* cecf2b178c28b0ae2d29fc0b3e0ae46bacc7b9d6

**Pull request merged**